pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +319,583 @@ func TestMemberService(t *testing.T) {
+					},
+				},
+				VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   "data",
+							Labels: newBasicStatefulSetLabels(),
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+							StorageClassName: newBasicRack().Storage.StorageClassName,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceStorage: resource.MustParse(newBasicRack().Storage.Capacity),
+								},
+							},
+						},
+					},
+				},
+				ServiceName:         "basic-client",
+				PodManagementPolicy: appsv1.OrderedReadyPodManagement,
+				UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+					Type: appsv1.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32Ptr(0),
+					},
+				},
+			},
+		}
+	}
+
+	newNodeAffinity := func() *corev1.NodeAffinity {
+		return &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "key",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"value"},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	newPodAffinity := func() *corev1.PodAffinity {
+		return &corev1.PodAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"key": "value",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	newPodAntiAffinity := func() *corev1.PodAntiAffinity {
+		return &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"key": "value",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	tt := []struct {
+		name                string
+		rack                scyllav1.RackSpec
+		scyllaCluster       *scyllav1.ScyllaCluster
+		existingStatefulSet *appsv1.StatefulSet
+		expectedStatefulSet *appsv1.StatefulSet
+		expectedError       error
+	}{
+		{
+			name:                "new StatefulSet",
+			rack:                newBasicRack(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			expectedStatefulSet: newBasicStatefulSet(),
+			expectedError:       nil,
+		},
+		{
+			name: "error for invalid Rack storage",
+			rack: func() scyllav1.RackSpec {
+				r := newBasicRack()
+				r.Storage = scyllav1.StorageSpec{
+					Capacity: "",
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			expectedStatefulSet: nil,
+			expectedError:       fmt.Errorf(`cannot parse "": %s`, resource.ErrFormatWrong),
```

Comment:
nit: errors print with `%v`

Reply:
or `%w`, guess that would be more fitting for this case

Reply:
thanks, fixed

Reply:
yeah, was afraid of that, but a code change in that direction would be valid too.